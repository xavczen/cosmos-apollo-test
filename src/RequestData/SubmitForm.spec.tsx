import toJson from 'enzyme-to-json';
import createTestContext from 'react-cosmos-test/enzyme';

import fixture from './SubmitForm.fixture';

const { mount, getWrapper, set } = createTestContext({ fixture });

describe('SubmitForm', () => {
  beforeEach(mount);

  it('renders correctly', () => {
    expect(toJson(getWrapper())).toMatchSnapshot();
  });

  it('should trigger mutation when submit button clicked', () => {
    const button = getWrapper('button');
    button.simulate('click');
    expect(toJson(getWrapper())).toMatchSnapshot();

    // TODO
  });
});
