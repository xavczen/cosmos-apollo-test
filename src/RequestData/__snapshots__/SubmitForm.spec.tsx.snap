// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitForm renders correctly 1`] = `
<SubmitForm>
  <div>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "saveExportDataRequest",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "input",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "input",
                          },
                        },
                      },
                    ],
                    "directives": Array [
                      Object {
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "type",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "ExportDataRequest",
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "path",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "api/v1/exports",
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "method",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "POST",
                            },
                          },
                        ],
                        "kind": "Directive",
                        "name": Object {
                          "kind": "Name",
                          "value": "rest",
                        },
                      },
                    ],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "saveExportDataRequest",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "NoResponse",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "ExportDataRequest",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 243,
            "start": 0,
          },
        }
      }
      update={[Function]}
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Submit
      </button>
    </Mutation>
  </div>
</SubmitForm>
`;

exports[`SubmitForm should trigger mutation when submit button clicked 1`] = `
<SubmitForm>
  <div>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "saveExportDataRequest",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "input",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "input",
                          },
                        },
                      },
                    ],
                    "directives": Array [
                      Object {
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "type",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "ExportDataRequest",
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "path",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "api/v1/exports",
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "method",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "POST",
                            },
                          },
                        ],
                        "kind": "Directive",
                        "name": Object {
                          "kind": "Name",
                          "value": "rest",
                        },
                      },
                    ],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "saveExportDataRequest",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "NoResponse",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "ExportDataRequest",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 243,
            "start": 0,
          },
        }
      }
      update={[Function]}
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Loading
      </button>
    </Mutation>
  </div>
</SubmitForm>
`;
